File: lib/components/widgets/gesture_detector.dart
import 'package:flutter/material.dart';
import 'package:flutter/gestures.dart';
import 'package:flutter/rendering.dart';
import '../base/flutter_component.dart';
import '../../core/virtual_dom_parser.dart';
import '../../core/component_registry.dart';
import '../../utils/gesture_parsers.dart';
import '../../utils/style_parsers/foundation/geometry_parsers.dart';

/// Flutter GestureDetector组件映射
/// 
/// 完全对齐Flutter GestureDetector API
/// 支持所有手势类型和回调
class GestureDetectorComponent extends FlutterComponent {
  @override
  String get componentType => 'GestureDetector';

  @override
  String get description => 'Flutter GestureDetector组件映射';

  @override
  Map<String, Type> get supportedProps => {
    // Tap手势
    'onTap': String,
    'onTapDown': String,
    'onTapUp': String,
    'onTapCancel': String,
    
    // DoubleTap手势
    'onDoubleTap': String,
    'onDoubleTapDown': String,
    'onDoubleTapCancel': String,
    
    // LongPress手势
    'onLongPress': String,
    'onLongPressDown': String,
    'onLongPressStart': String,
    'onLongPressMoveUpdate': String,
    'onLongPressUp': String,
    'onLongPressEnd': String,
    'onLongPressCancel': String,
    
    // SecondaryTap手势
    'onSecondaryTap': String,
    'onSecondaryTapDown': String,
    'onSecondaryTapUp': String,
    'onSecondaryTapCancel': String,
    
    // TertiaryTap手势
    'onTertiaryTap': String,
    'onTertiaryTapDown': String,
    'onTertiaryTapUp': String,
    'onTertiaryTapCancel': String,
    
    // SecondaryLongPress手势
    'onSecondaryLongPress': String,
    'onSecondaryLongPressDown': String,
    'onSecondaryLongPressStart': String,
    'onSecondaryLongPressMoveUpdate': String,
    'onSecondaryLongPressUp': String,
    'onSecondaryLongPressEnd': String,
    
    // TertiaryLongPress手势
    'onTertiaryLongPress': String,
    'onTertiaryLongPressDown': String,
    'onTertiaryLongPressStart': String,
    'onTertiaryLongPressMoveUpdate': String,
    'onTertiaryLongPressUp': String,
    'onTertiaryLongPressEnd': String,
    
    // Pan手势
    'onPanDown': String,
    'onPanStart': String,
    'onPanUpdate': String,
    'onPanEnd': String,
    'onPanCancel': String,
    
    // HorizontalDrag手势
    'onHorizontalDragDown': String,
    'onHorizontalDragStart': String,
    'onHorizontalDragUpdate': String,
    'onHorizontalDragEnd': String,
    'onHorizontalDragCancel': String,
    
    // VerticalDrag手势
    'onVerticalDragDown': String,
    'onVerticalDragStart': String,
    'onVerticalDragUpdate': String,
    'onVerticalDragEnd': String,
    'onVerticalDragCancel': String,
    
    // Scale手势
    'onScaleStart': String,
    'onScaleUpdate': String,
    'onScaleEnd': String,
    
    // ForcePress手势
    'onForcePressStart': String,
    'onForcePressPeak': String,
    'onForcePressUpdate': String,
    'onForcePressEnd': String,
    
    // 行为控制
    'behavior': String,
    'excludeFromSemantics': bool,
    'dragStartBehavior': String,
    'trackpadScrollCausesScale': bool,
    'trackpadScrollToScaleFactor': num,
    
    'id': String,
  };

  @override
  List<String> get supportedEvents => [
    'onTap', 'onTapDown', 'onTapUp', 'onTapCancel',
    'onDoubleTap', 'onDoubleTapDown', 'onDoubleTapCancel',
    'onLongPress', 'onLongPressDown', 'onLongPressStart', 'onLongPressMoveUpdate', 'onLongPressUp', 'onLongPressEnd', 'onLongPressCancel',
    'onSecondaryTap', 'onSecondaryTapDown', 'onSecondaryTapUp', 'onSecondaryTapCancel',
    'onTertiaryTap', 'onTertiaryTapDown', 'onTertiaryTapUp', 'onTertiaryTapCancel',
    'onSecondaryLongPress', 'onSecondaryLongPressDown', 'onSecondaryLongPressStart', 'onSecondaryLongPressMoveUpdate', 'onSecondaryLongPressUp', 'onSecondaryLongPressEnd',
    'onTertiaryLongPress', 'onTertiaryLongPressDown', 'onTertiaryLongPressStart', 'onTertiaryLongPressMoveUpdate', 'onTertiaryLongPressUp', 'onTertiaryLongPressEnd',
    'onPanDown', 'onPanStart', 'onPanUpdate', 'onPanEnd', 'onPanCancel',
    'onHorizontalDragDown', 'onHorizontalDragStart', 'onHorizontalDragUpdate', 'onHorizontalDragEnd', 'onHorizontalDragCancel',
    'onVerticalDragDown', 'onVerticalDragStart', 'onVerticalDragUpdate', 'onVerticalDragEnd', 'onVerticalDragCancel',
    'onScaleStart', 'onScaleUpdate', 'onScaleEnd',
    'onForcePressStart', 'onForcePressPeak', 'onForcePressUpdate', 'onForcePressEnd',
  ];

  @override
  bool get supportsChildren => true;

  @override
  Widget build(VirtualDOM vdom) {
    final id = vdom.getProp<String>('id');
    
    // 解析行为控制属性
    final behavior = _parseHitTestBehavior(vdom.getProp<String>('behavior'));
    final excludeFromSemantics = vdom.getProp<bool>('excludeFromSemantics') ?? false;
    final dragStartBehavior = _parseDragStartBehavior(vdom.getProp<String>('dragStartBehavior')) ?? DragStartBehavior.start;
    
    // 获取所有事件ID
    final onTap = vdom.getProp<String>('onTap');
    final onTapDown = vdom.getProp<String>('onTapDown');
    final onTapUp = vdom.getProp<String>('onTapUp');
    final onTapCancel = vdom.getProp<String>('onTapCancel');
    
    final onDoubleTap = vdom.getProp<String>('onDoubleTap');
    final onDoubleTapDown = vdom.getProp<String>('onDoubleTapDown');
    final onDoubleTapCancel = vdom.getProp<String>('onDoubleTapCancel');
    
    final onLongPress = vdom.getProp<String>('onLongPress');
    final onLongPressDown = vdom.getProp<String>('onLongPressDown');
    final onLongPressStart = vdom.getProp<String>('onLongPressStart');
    final onLongPressMoveUpdate = vdom.getProp<String>('onLongPressMoveUpdate');
    final onLongPressUp = vdom.getProp<String>('onLongPressUp');
    final onLongPressEnd = vdom.getProp<String>('onLongPressEnd');
    final onLongPressCancel = vdom.getProp<String>('onLongPressCancel');
    
    final onSecondaryTap = vdom.getProp<String>('onSecondaryTap');
    final onSecondaryTapDown = vdom.getProp<String>('onSecondaryTapDown');
    final onSecondaryTapUp = vdom.getProp<String>('onSecondaryTapUp');
    final onSecondaryTapCancel = vdom.getProp<String>('onSecondaryTapCancel');
    
    final onTertiaryTapDown = vdom.getProp<String>('onTertiaryTapDown');
    final onTertiaryTapUp = vdom.getProp<String>('onTertiaryTapUp');
    final onTertiaryTapCancel = vdom.getProp<String>('onTertiaryTapCancel');
    
    final onSecondaryLongPress = vdom.getProp<String>('onSecondaryLongPress');
    final onSecondaryLongPressDown = vdom.getProp<String>('onSecondaryLongPressDown');
    final onSecondaryLongPressStart = vdom.getProp<String>('onSecondaryLongPressStart');
    final onSecondaryLongPressMoveUpdate = vdom.getProp<String>('onSecondaryLongPressMoveUpdate');
    final onSecondaryLongPressUp = vdom.getProp<String>('onSecondaryLongPressUp');
    final onSecondaryLongPressEnd = vdom.getProp<String>('onSecondaryLongPressEnd');
    
    final onTertiaryLongPress = vdom.getProp<String>('onTertiaryLongPress');
    final onTertiaryLongPressDown = vdom.getProp<String>('onTertiaryLongPressDown');
    final onTertiaryLongPressStart = vdom.getProp<String>('onTertiaryLongPressStart');
    final onTertiaryLongPressMoveUpdate = vdom.getProp<String>('onTertiaryLongPressMoveUpdate');
    final onTertiaryLongPressUp = vdom.getProp<String>('onTertiaryLongPressUp');
    final onTertiaryLongPressEnd = vdom.getProp<String>('onTertiaryLongPressEnd');
    
    final onPanDown = vdom.getProp<String>('onPanDown');
    final onPanStart = vdom.getProp<String>('onPanStart');
    final onPanUpdate = vdom.getProp<String>('onPanUpdate');
    final onPanEnd = vdom.getProp<String>('onPanEnd');
    final onPanCancel = vdom.getProp<String>('onPanCancel');
    
    final onHorizontalDragDown = vdom.getProp<String>('onHorizontalDragDown');
    final onHorizontalDragStart = vdom.getProp<String>('onHorizontalDragStart');
    final onHorizontalDragUpdate = vdom.getProp<String>('onHorizontalDragUpdate');
    final onHorizontalDragEnd = vdom.getProp<String>('onHorizontalDragEnd');
    final onHorizontalDragCancel = vdom.getProp<String>('onHorizontalDragCancel');
    
    final onVerticalDragDown = vdom.getProp<String>('onVerticalDragDown');
    final onVerticalDragStart = vdom.getProp<String>('onVerticalDragStart');
    final onVerticalDragUpdate = vdom.getProp<String>('onVerticalDragUpdate');
    final onVerticalDragEnd = vdom.getProp<String>('onVerticalDragEnd');
    final onVerticalDragCancel = vdom.getProp<String>('onVerticalDragCancel');
    
    final onScaleStart = vdom.getProp<String>('onScaleStart');
    final onScaleUpdate = vdom.getProp<String>('onScaleUpdate');
    final onScaleEnd = vdom.getProp<String>('onScaleEnd');
    
    final onForcePressStart = vdom.getProp<String>('onForcePressStart');
    final onForcePressPeak = vdom.getProp<String>('onForcePressPeak');
    final onForcePressUpdate = vdom.getProp<String>('onForcePressUpdate');
    final onForcePressEnd = vdom.getProp<String>('onForcePressEnd');
    
    // 获取子组件
    final childVdomList = vdom.getChildrenList();
    Widget? child;
    if (childVdomList.isNotEmpty) {
      child = ComponentRegistry.instance.buildComponent(childVdomList.first);
    }
    
    return GestureDetector(
      key: id != null ? Key(id) : null,
      
      // Tap手势
      onTap: onTap != null ? () => ComponentRegistry.instance.triggerEvent(onTap, null) : null,
      onTapDown: onTapDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onTapDown, GestureParsers.tapDownDetailsToJson(details)) : null,
      onTapUp: onTapUp != null ? (details) => ComponentRegistry.instance.triggerEvent(onTapUp, GestureParsers.tapUpDetailsToJson(details)) : null,
      onTapCancel: onTapCancel != null ? () => ComponentRegistry.instance.triggerEvent(onTapCancel, null) : null,
      
      // DoubleTap手势
      onDoubleTap: onDoubleTap != null ? () => ComponentRegistry.instance.triggerEvent(onDoubleTap, null) : null,
      onDoubleTapDown: onDoubleTapDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onDoubleTapDown, GestureParsers.tapDownDetailsToJson(details)) : null,
      onDoubleTapCancel: onDoubleTapCancel != null ? () => ComponentRegistry.instance.triggerEvent(onDoubleTapCancel, null) : null,
      
      // LongPress手势
      onLongPress: onLongPress != null ? () => ComponentRegistry.instance.triggerEvent(onLongPress, null) : null,
      onLongPressDown: onLongPressDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onLongPressDown, GestureParsers.longPressDownDetailsToJson(details)) : null,
      onLongPressStart: onLongPressStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onLongPressStart, GestureParsers.longPressStartDetailsToJson(details)) : null,
      onLongPressMoveUpdate: onLongPressMoveUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onLongPressMoveUpdate, GestureParsers.longPressMoveUpdateDetailsToJson(details)) : null,
      onLongPressUp: onLongPressUp != null ? () => ComponentRegistry.instance.triggerEvent(onLongPressUp, null) : null,
      onLongPressEnd: onLongPressEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onLongPressEnd, GestureParsers.longPressEndDetailsToJson(details)) : null,
      onLongPressCancel: onLongPressCancel != null ? () => ComponentRegistry.instance.triggerEvent(onLongPressCancel, null) : null,
      
      // SecondaryTap手势
      onSecondaryTap: onSecondaryTap != null ? () => ComponentRegistry.instance.triggerEvent(onSecondaryTap, null) : null,
      onSecondaryTapDown: onSecondaryTapDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryTapDown, GestureParsers.tapDownDetailsToJson(details)) : null,
      onSecondaryTapUp: onSecondaryTapUp != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryTapUp, GestureParsers.tapUpDetailsToJson(details)) : null,
      onSecondaryTapCancel: onSecondaryTapCancel != null ? () => ComponentRegistry.instance.triggerEvent(onSecondaryTapCancel, null) : null,
      
      // TertiaryTap手势
      onTertiaryTapDown: onTertiaryTapDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryTapDown, GestureParsers.tapDownDetailsToJson(details)) : null,
      onTertiaryTapUp: onTertiaryTapUp != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryTapUp, GestureParsers.tapUpDetailsToJson(details)) : null,
      onTertiaryTapCancel: onTertiaryTapCancel != null ? () => ComponentRegistry.instance.triggerEvent(onTertiaryTapCancel, null) : null,
      
      // SecondaryLongPress手势
      onSecondaryLongPress: onSecondaryLongPress != null ? () => ComponentRegistry.instance.triggerEvent(onSecondaryLongPress, null) : null,
      onSecondaryLongPressDown: onSecondaryLongPressDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryLongPressDown, GestureParsers.longPressDownDetailsToJson(details)) : null,
      onSecondaryLongPressStart: onSecondaryLongPressStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryLongPressStart, GestureParsers.longPressStartDetailsToJson(details)) : null,
      onSecondaryLongPressMoveUpdate: onSecondaryLongPressMoveUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryLongPressMoveUpdate, GestureParsers.longPressMoveUpdateDetailsToJson(details)) : null,
      onSecondaryLongPressUp: onSecondaryLongPressUp != null ? () => ComponentRegistry.instance.triggerEvent(onSecondaryLongPressUp, null) : null,
      onSecondaryLongPressEnd: onSecondaryLongPressEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onSecondaryLongPressEnd, GestureParsers.longPressEndDetailsToJson(details)) : null,
      
      // TertiaryLongPress手势
      onTertiaryLongPress: onTertiaryLongPress != null ? () => ComponentRegistry.instance.triggerEvent(onTertiaryLongPress, null) : null,
      onTertiaryLongPressDown: onTertiaryLongPressDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryLongPressDown, GestureParsers.longPressDownDetailsToJson(details)) : null,
      onTertiaryLongPressStart: onTertiaryLongPressStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryLongPressStart, GestureParsers.longPressStartDetailsToJson(details)) : null,
      onTertiaryLongPressMoveUpdate: onTertiaryLongPressMoveUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryLongPressMoveUpdate, GestureParsers.longPressMoveUpdateDetailsToJson(details)) : null,
      onTertiaryLongPressUp: onTertiaryLongPressUp != null ? () => ComponentRegistry.instance.triggerEvent(onTertiaryLongPressUp, null) : null,
      onTertiaryLongPressEnd: onTertiaryLongPressEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onTertiaryLongPressEnd, GestureParsers.longPressEndDetailsToJson(details)) : null,
      
      // Pan手势
      onPanDown: onPanDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onPanDown, GestureParsers.dragDownDetailsToJson(details)) : null,
      onPanStart: onPanStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onPanStart, GestureParsers.dragStartDetailsToJson(details)) : null,
      onPanUpdate: onPanUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onPanUpdate, GestureParsers.dragUpdateDetailsToJson(details)) : null,
      onPanEnd: onPanEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onPanEnd, GestureParsers.dragEndDetailsToJson(details)) : null,
      onPanCancel: onPanCancel != null ? () => ComponentRegistry.instance.triggerEvent(onPanCancel, null) : null,
      
      // HorizontalDrag手势
      onHorizontalDragDown: onHorizontalDragDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onHorizontalDragDown, GestureParsers.dragDownDetailsToJson(details)) : null,
      onHorizontalDragStart: onHorizontalDragStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onHorizontalDragStart, GestureParsers.dragStartDetailsToJson(details)) : null,
      onHorizontalDragUpdate: onHorizontalDragUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onHorizontalDragUpdate, GestureParsers.dragUpdateDetailsToJson(details)) : null,
      onHorizontalDragEnd: onHorizontalDragEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onHorizontalDragEnd, GestureParsers.dragEndDetailsToJson(details)) : null,
      onHorizontalDragCancel: onHorizontalDragCancel != null ? () => ComponentRegistry.instance.triggerEvent(onHorizontalDragCancel, null) : null,
      
      // VerticalDrag手势
      onVerticalDragDown: onVerticalDragDown != null ? (details) => ComponentRegistry.instance.triggerEvent(onVerticalDragDown, GestureParsers.dragDownDetailsToJson(details)) : null,
      onVerticalDragStart: onVerticalDragStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onVerticalDragStart, GestureParsers.dragStartDetailsToJson(details)) : null,
      onVerticalDragUpdate: onVerticalDragUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onVerticalDragUpdate, GestureParsers.dragUpdateDetailsToJson(details)) : null,
      onVerticalDragEnd: onVerticalDragEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onVerticalDragEnd, GestureParsers.dragEndDetailsToJson(details)) : null,
      onVerticalDragCancel: onVerticalDragCancel != null ? () => ComponentRegistry.instance.triggerEvent(onVerticalDragCancel, null) : null,
      
      // Scale手势
      onScaleStart: onScaleStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onScaleStart, GestureParsers.scaleStartDetailsToJson(details)) : null,
      onScaleUpdate: onScaleUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onScaleUpdate, GestureParsers.scaleUpdateDetailsToJson(details)) : null,
      onScaleEnd: onScaleEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onScaleEnd, GestureParsers.scaleEndDetailsToJson(details)) : null,
      
      // ForcePress手势
      onForcePressStart: onForcePressStart != null ? (details) => ComponentRegistry.instance.triggerEvent(onForcePressStart, GestureParsers.forcePressDetailsToJson(details)) : null,
      onForcePressPeak: onForcePressPeak != null ? (details) => ComponentRegistry.instance.triggerEvent(onForcePressPeak, GestureParsers.forcePressDetailsToJson(details)) : null,
      onForcePressUpdate: onForcePressUpdate != null ? (details) => ComponentRegistry.instance.triggerEvent(onForcePressUpdate, GestureParsers.forcePressDetailsToJson(details)) : null,
      onForcePressEnd: onForcePressEnd != null ? (details) => ComponentRegistry.instance.triggerEvent(onForcePressEnd, GestureParsers.forcePressDetailsToJson(details)) : null,
      
      // 行为控制
      behavior: behavior,
      excludeFromSemantics: excludeFromSemantics,
      dragStartBehavior: dragStartBehavior,
      
      child: child,
    );
  }

  /// 解析HitTestBehavior
  static HitTestBehavior? _parseHitTestBehavior(String? value) {
    if (value == null) return null;
    switch (value) {
      case 'deferToChild':
        return HitTestBehavior.deferToChild;
      case 'opaque':
        return HitTestBehavior.opaque;
      case 'translucent':
        return HitTestBehavior.translucent;
      default:
        return null;
    }
  }

  /// 解析DragStartBehavior
  static DragStartBehavior? _parseDragStartBehavior(String? value) {
    if (value == null) return null;
    switch (value) {
      case 'down':
        return DragStartBehavior.down;
      case 'start':
        return DragStartBehavior.start;
      default:
        return null;
    }
  }
}

File: packages/components/src/gesture-detector.ts
/**
 * Flutter GestureDetector组件的React实现
 * 完全对齐Flutter GestureDetector API
 */

import * as React from 'react';
import type {
  TapDownDetails,
  TapUpDetails,
  DragStartDetails,
  DragUpdateDetails,
  DragEndDetails,
  DragDownDetails,
  ScaleStartDetails,
  ScaleUpdateDetails,
  ScaleEndDetails,
  LongPressStartDetails,
  LongPressMoveUpdateDetails,
  LongPressEndDetails,
  ForcePressDetails,
  HitTestBehaviorValue,
  DragStartBehaviorValue,
  PointerDeviceKindValue,
} from './styles/types';

export interface GestureDetectorProps {
  children?: React.ReactNode;
  
  // ===== Tap手势 =====
  /** 单击回调 */
  onTap?: () => void;
  /** 单击按下回调 */
  onTapDown?: (details: TapDownDetails) => void;
  /** 单击抬起回调 */
  onTapUp?: (details: TapUpDetails) => void;
  /** 单击取消回调 */
  onTapCancel?: () => void;
  
  // ===== DoubleTap双击手势 =====
  /** 双击回调 */
  onDoubleTap?: () => void;
  /** 双击按下回调 */
  onDoubleTapDown?: (details: TapDownDetails) => void;
  /** 双击取消回调 */
  onDoubleTapCancel?: () => void;
  
  // ===== LongPress长按手势 =====
  /** 长按回调 */
  onLongPress?: () => void;
  /** 长按开始回调 */
  onLongPressStart?: (details: LongPressStartDetails) => void;
  /** 长按移动更新回调 */
  onLongPressMoveUpdate?: (details: LongPressMoveUpdateDetails) => void;
  /** 长按抬起回调 */
  onLongPressUp?: () => void;
  /** 长按结束回调 */
  onLongPressEnd?: (details: LongPressEndDetails) => void;
  /** 长按取消回调 */
  onLongPressCancel?: () => void;
  
  // ===== Secondary Tap次要点击（右键）手势 =====
  /** 次要点击按下回调 */
  onSecondaryTap?: () => void;
  /** 次要点击按下回调 */
  onSecondaryTapDown?: (details: TapDownDetails) => void;
  /** 次要点击抬起回调 */
  onSecondaryTapUp?: (details: TapUpDetails) => void;
  /** 次要点击取消回调 */
  onSecondaryTapCancel?: () => void;
  
  // ===== Tertiary Tap第三按键点击（中键）手势 =====
  /** 第三按键点击按下回调 */
  onTertiaryTapDown?: (details: TapDownDetails) => void;
  /** 第三按键点击抬起回调 */
  onTertiaryTapUp?: (details: TapUpDetails) => void;
  /** 第三按键点击取消回调 */
  onTertiaryTapCancel?: () => void;
  
  // ===== Secondary LongPress次要长按（右键长按）手势 =====
  /** 次要长按回调 */
  onSecondaryLongPress?: () => void;
  /** 次要长按开始回调 */
  onSecondaryLongPressStart?: (details: LongPressStartDetails) => void;
  /** 次要长按移动更新回调 */
  onSecondaryLongPressMoveUpdate?: (details: LongPressMoveUpdateDetails) => void;
  /** 次要长按抬起回调 */
  onSecondaryLongPressUp?: () => void;
  /** 次要长按结束回调 */
  onSecondaryLongPressEnd?: (details: LongPressEndDetails) => void;
  
  // ===== Tertiary LongPress第三按键长按（中键长按）手势 =====
  /** 第三按键长按回调 */
  onTertiaryLongPress?: () => void;
  /** 第三按键长按开始回调 */
  onTertiaryLongPressStart?: (details: LongPressStartDetails) => void;
  /** 第三按键长按移动更新回调 */
  onTertiaryLongPressMoveUpdate?: (details: LongPressMoveUpdateDetails) => void;
  /** 第三按键长按抬起回调 */
  onTertiaryLongPressUp?: () => void;
  /** 第三按键长按结束回调 */
  onTertiaryLongPressEnd?: (details: LongPressEndDetails) => void;
  
  // ===== Pan拖拽手势（同时支持水平和垂直）=====
  /** 拖拽按下回调 */
  onPanDown?: (details: DragDownDetails) => void;
  /** 拖拽开始回调 */
  onPanStart?: (details: DragStartDetails) => void;
  /** 拖拽更新回调 */
  onPanUpdate?: (details: DragUpdateDetails) => void;
  /** 拖拽结束回调 */
  onPanEnd?: (details: DragEndDetails) => void;
  /** 拖拽取消回调 */
  onPanCancel?: () => void;
  
  // ===== HorizontalDrag水平拖拽手势 =====
  /** 水平拖拽按下回调 */
  onHorizontalDragDown?: (details: DragDownDetails) => void;
  /** 水平拖拽开始回调 */
  onHorizontalDragStart?: (details: DragStartDetails) => void;
  /** 水平拖拽更新回调 */
  onHorizontalDragUpdate?: (details: DragUpdateDetails) => void;
  /** 水平拖拽结束回调 */
  onHorizontalDragEnd?: (details: DragEndDetails) => void;
  /** 水平拖拽取消回调 */
  onHorizontalDragCancel?: () => void;
  
  // ===== VerticalDrag垂直拖拽手势 =====
  /** 垂直拖拽按下回调 */
  onVerticalDragDown?: (details: DragDownDetails) => void;
  /** 垂直拖拽开始回调 */
  onVerticalDragStart?: (details: DragStartDetails) => void;
  /** 垂直拖拽更新回调 */
  onVerticalDragUpdate?: (details: DragUpdateDetails) => void;
  /** 垂直拖拽结束回调 */
  onVerticalDragEnd?: (details: DragEndDetails) => void;
  /** 垂直拖拽取消回调 */
  onVerticalDragCancel?: () => void;
  
  // ===== Scale缩放/旋转手势 =====
  /** 缩放开始回调 */
  onScaleStart?: (details: ScaleStartDetails) => void;
  /** 缩放更新回调 */
  onScaleUpdate?: (details: ScaleUpdateDetails) => void;
  /** 缩放结束回调 */
  onScaleEnd?: (details: ScaleEndDetails) => void;
  
  // ===== ForcePress力度按压手势（3D Touch / Haptic Touch）=====
  /** 力度按压开始回调 */
  onForcePressStart?: (details: ForcePressDetails) => void;
  /** 力度按压峰值回调 */
  onForcePressPeak?: (details: ForcePressDetails) => void;
  /** 力度按压更新回调 */
  onForcePressUpdate?: (details: ForcePressDetails) => void;
  /** 力度按压结束回调 */
  onForcePressEnd?: (details: ForcePressDetails) => void;
  
  // ===== 行为控制属性 =====
  /** 命中测试行为 */
  behavior?: HitTestBehaviorValue;
  /** 是否排除无障碍树 */
  excludeFromSemantics?: boolean;
  /** 拖拽开始行为 */
  dragStartBehavior?: DragStartBehaviorValue;
  /** 触控板滚动是否触发缩放手势 */
  trackpadScrollCausesScale?: boolean;
  /** 触控板滚动到缩放的转换因子 */
  trackpadScrollToScaleFactor?: number;
  /** 支持的设备类型列表 */
  supportedDevices?: PointerDeviceKindValue[];
  
  /** 唯一标识（映射为Flutter Key） */
  id?: string;
}

export const GestureDetector: React.FC<GestureDetectorProps> = ({
  children,
  
  // Tap手势
  onTap,
  onTapDown,
  onTapUp,
  onTapCancel,
  
  // DoubleTap手势
  onDoubleTap,
  onDoubleTapDown,
  onDoubleTapCancel,
  
  // LongPress手势
  onLongPress,
  onLongPressStart,
  onLongPressMoveUpdate,
  onLongPressUp,
  onLongPressEnd,
  onLongPressCancel,
  
  // SecondaryTap手势
  onSecondaryTap,
  onSecondaryTapDown,
  onSecondaryTapUp,
  onSecondaryTapCancel,
  
  // TertiaryTap手势
  onTertiaryTapDown,
  onTertiaryTapUp,
  onTertiaryTapCancel,
  
  // SecondaryLongPress手势
  onSecondaryLongPress,
  onSecondaryLongPressStart,
  onSecondaryLongPressMoveUpdate,
  onSecondaryLongPressUp,
  onSecondaryLongPressEnd,
  
  // TertiaryLongPress手势
  onTertiaryLongPress,
  onTertiaryLongPressStart,
  onTertiaryLongPressMoveUpdate,
  onTertiaryLongPressUp,
  onTertiaryLongPressEnd,
  
  // Pan手势
  onPanDown,
  onPanStart,
  onPanUpdate,
  onPanEnd,
  onPanCancel,
  
  // HorizontalDrag手势
  onHorizontalDragDown,
  onHorizontalDragStart,
  onHorizontalDragUpdate,
  onHorizontalDragEnd,
  onHorizontalDragCancel,
  
  // VerticalDrag手势
  onVerticalDragDown,
  onVerticalDragStart,
  onVerticalDragUpdate,
  onVerticalDragEnd,
  onVerticalDragCancel,
  
  // Scale手势
  onScaleStart,
  onScaleUpdate,
  onScaleEnd,
  
  // ForcePress手势
  onForcePressStart,
  onForcePressPeak,
  onForcePressUpdate,
  onForcePressEnd,
  
  // 行为控制
  behavior,
  excludeFromSemantics,
  dragStartBehavior,
  trackpadScrollCausesScale,
  trackpadScrollToScaleFactor,
  supportedDevices,
  
  id,
  ...props
}) => {
  return React.createElement('GestureDetector', {
    // Tap手势
    onTap,
    onTapDown,
    onTapUp,
    onTapCancel,
    
    // DoubleTap手势
    onDoubleTap,
    onDoubleTapDown,
    onDoubleTapCancel,
    
    // LongPress手势
    onLongPress,
    onLongPressStart,
    onLongPressMoveUpdate,
    onLongPressUp,
    onLongPressEnd,
    onLongPressCancel,
    
    // SecondaryTap手势
    onSecondaryTap,
    onSecondaryTapDown,
    onSecondaryTapUp,
    onSecondaryTapCancel,
    
    // TertiaryTap手势
    onTertiaryTapDown,
    onTertiaryTapUp,
    onTertiaryTapCancel,
    
    // SecondaryLongPress手势
    onSecondaryLongPress,
    onSecondaryLongPressStart,
    onSecondaryLongPressMoveUpdate,
    onSecondaryLongPressUp,
    onSecondaryLongPressEnd,
    
    // TertiaryLongPress手势
    onTertiaryLongPress,
    onTertiaryLongPressStart,
    onTertiaryLongPressMoveUpdate,
    onTertiaryLongPressUp,
    onTertiaryLongPressEnd,
    
    // Pan手势
    onPanDown,
    onPanStart,
    onPanUpdate,
    onPanEnd,
    onPanCancel,
    
    // HorizontalDrag手势
    onHorizontalDragDown,
    onHorizontalDragStart,
    onHorizontalDragUpdate,
    onHorizontalDragEnd,
    onHorizontalDragCancel,
    
    // VerticalDrag手势
    onVerticalDragDown,
    onVerticalDragStart,
    onVerticalDragUpdate,
    onVerticalDragEnd,
    onVerticalDragCancel,
    
    // Scale手势
    onScaleStart,
    onScaleUpdate,
    onScaleEnd,
    
    // ForcePress手势
    onForcePressStart,
    onForcePressPeak,
    onForcePressUpdate,
    onForcePressEnd,
    
    // 行为控制
    behavior,
    excludeFromSemantics,
    dragStartBehavior,
    trackpadScrollCausesScale,
    trackpadScrollToScaleFactor,
    supportedDevices,
    
    id,
    ...props
  }, children);
};
